backend/tracim_backend/lib/core/workspace.py

```diff
@@ -261,6 +261,32 @@ def execute_created_workspace_actions(self, workspace: Workspace) -> None:
                 except AgendaServerConnectionError as exc:
                     logger.error(self, 'Cannot connect to agenda server')
                     logger.exception(self, exc)
+                except Exception as exc:
+                    logger.error(self, 'Something goes wrong during agenda create/update')
+                    logger.exception(self, exc)
+
+    def execute_update_workspace_actions(self, workspace: Workspace) -> None:
+        from tracim_backend.lib.agenda.agenda import AgendaApi
+        if self._config.CALDAV_ENABLED:
+            agenda_api = AgendaApi(
+                current_user = self._user,
+                session = self._session,
+                config = self._config
+            )
+            if workspace.agenda_enabled:
+                try:
+                    agenda_already_exist = agenda_api.ensure_workspace_agenda_exist(workspace)
+                    if agenda_already_exist:
+                        logger.warning(
+                            self,
+                            'workspace {} is just created but it own agenda already exist !!'.format(workspace.user_id)
+                        )
+                except AgendaServerConnectionError as exc:
+                    logger.error(self, 'Cannot connect to agenda server')
+                    logger.exception(self, exc)
```

Comment:
wath is `exception()` method? this is probably `log_exception()`, no? could you rename this? (or its not our code?)
By the way, what is the goal of logging the exception in this case: the error is clearly identified

Reply:
It's just same as logger.exception() method from https://docs.python.org/3.7/library/logging.html .
As we override logger by our own class we can rename it but that a bit weird to have similar naming for some method and different for others.

Reply:
you're right, it's a wrong idea to rename it. I do not understand the interest of this method compared to using `logger.error(xxx)`

Reply:
@lebouquetin  logger.exception do print exception on logging, logger.error is just a error level message. For this case, we can effectively not logging the exception but "connection error" give not lot of information. 
it can mean multiple things: see http://docs.python-requests.org/en/master/_modules/requests/exceptions/ (we do a except requests.exceptions.ConnectionError as exc) . Maybe should we be more precise.

Reply:
- I think you should at least complete the error log with address the service tries to connect to.
- Keep the logger.exception

Reply:
will fix later #1592